packages/stats/gbstats/gbstats.py

```diff
@@ -142,73 +153,89 @@ def reduce_dimensionality(df, max=20):
-        n_a = s["baseline_users"]
-        m_a = s["baseline_mean"]
-        x_a = s["baseline_count"]
-        s_a = s["baseline_stddev"]
-        s["baseline_cr"] = m_a
+        stat_a = Statistic(
+            value=s["baseline_mean"],
+            stddev=s["baseline_stddev"],
+            n=s["baseline_users"],
+            count=s["baseline_count"],
+        )
+        s["baseline_cr"] = stat_a.value
 
         # List of users in each variation (used for SRM check)
         users = [0] * num_variations
-        users[0] = n_a
+        users[0] = stat_a.n
 
         # Loop through each non-baseline variation and run an analysis
         baseline_risk = 0
         for i in range(1, num_variations):
             # Variation values
-            n_b = s[f"v{i}_users"]
-            m_b = s[f"v{i}_mean"]
-            x_b = s[f"v{i}_count"]
-            s_b = s[f"v{i}_stddev"]
+            stat_b = Statistic(
+                value=s[f"v{i}_mean"],
+                stddev=s[f"v{i}_stddev"],
+                n=s[f"v{i}_users"],
+                count=s[f"v{i}_count"],
+            )
 
-            s[f"v{i}_cr"] = m_b
-            s[f"v{i}_expected"] = (m_b / m_a) - 1 if m_a > 0 else 0
+            s[f"v{i}_cr"] = stat_b.value
+            s[f"v{i}_expected"] = (
+                (stat_b.value / stat_a.value) - 1 if stat_a.value > 0 else 0
+            )
 
-            users[i] = n_b
+            users[i] = stat_b.n
 
             # Run the A/B test analysis of baseline vs variation
-            if type == "binomial":
-                res = binomial_ab_test(m_a * x_a, x_a, m_b * x_b, x_b)
+            if engine == StatsEngine.BAYESIAN:
+                if type == "binomial":
+                    # TODO consider refactor so binomial test just takes proportions
+                    test: BaseABTest = BinomialBayesianABTest(
+                        stat_a, stat_b, inverse=inverse
+                    )
+                else:
+                    test: BaseABTest = GaussianBayesianABTest(
+                        stat_a, stat_b, inverse=inverse
+                    )
+
+                res: TestResult = test.compute_result()
+
+                # The baseline risk is the max risk of any of the variation A/B tests
+                if res.relative_risk[0] > baseline_risk:
+                    baseline_risk = res.relative_risk[0]
+
+                s.at[f"v{i}_rawrisk"] = res.risk
+                s[f"v{i}_risk"] = res.relative_risk[1]
+                s[f"v{i}_prob_beat_baseline"] = res.chance_to_win
             else:
-                res = gaussian_ab_test(m_a, s_a, n_a, m_b, s_b, n_b)
-
-            # Flip risk and chance to win for inverse metrics
-            risk0 = res["risk"][0] if not inverse else res["risk"][1]
-            risk1 = res["risk"][1] if not inverse else res["risk"][0]
-            ctw = res["chance_to_win"] if not inverse else 1 - res["chance_to_win"]
+                test: BaseABTest = TwoSidedTTest(stat_a, stat_b)
```

Comment:
Right now the frequentist test has no knowledge of the direction (inverse) of the test, and this will need to be remedied in a follow-up PR unless we can instead use the metric context in the results UI to help render the results in the appropriate manner.